README.md

```diff
@@ -5,7 +5,99 @@
 [![LICENSE](https://img.shields.io/github/license/leapfrogtechnology/async-store.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store/blob/master/LICENSE)
 [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://github.com/leapfrogtechnology/async-store)
 
-Global store utility for an async operation lifecycle and chain of callbacks. 
+Global store utility for an async operation lifecycle and chain of callbacks. It is utility tool similar to [continuation-local-storage](https://github.com/othiym23/node-continuation-local-storage) which allows us to set and get values that are scoped to the lifetime of these chains of callbacks. It uses [domain](https://nodejs.org/api/domain.html) native Node.js module.
+
+**Simple example,**
+
+```js
+var asyncStore = require('async-store');
+
+var callback = function {
+  asyncStore({ foo: 'Hello', bar: 'World' });
+
+  Promise.resolve()
+    .then(() => {
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      console.log('Value of foo: ', globalStore.get('foo'));
+    })
+    .then(() => {
+      // Store value is also available at the end of the promise chain.
+      console.log('Value of foo: ', globalStore.get('foo'));
+    });
+};
+
+asyncStore.initialize()(callback);
+```
+
+**Simple express.js example or [here](https://github.com/kabirbaidhya/async-store-sample)**
```

Comment:
1. Heading - `Express Example`
2. Move the other example in a different section `More Examples` under which you can just put a link to the source.